src/sql/workbench/services/restore/browser/restoreDialog.ts

```diff
@@ -696,19 +870,38 @@ export class RestoreDialog extends Modal {
 
 	private resetDialog(): void {
 		this.hideError();
-		this._restoreFromSelectBox!.selectWithOptionName(this._databaseTitle);
-		this.onRestoreFromChanged(this._databaseTitle);
+		if (!this._isManagedInstance) {
+			this._restoreFromSelectBox!.selectWithOptionName(this._databaseTitle);
+			this.onRestoreFromChanged(this._databaseTitle);
+		}
 		this._sourceDatabaseSelectBox!.select(0);
-		this._panel!.showTab(this._generalTabId!);
+		this._panel!.showTab(this._generalTab.identifier!);
 		this._isBackupFileCheckboxChanged = false;
 		this.removeErrorMessage();
 		this.resetRestoreContent();
 	}
 
-	public open(serverName: string, ownerUri: string) {
+	public open(serverName: string, ownerUri: string, isManagedInstance: boolean) {
```

Comment:
Just pass in the engine edition - we should stive to avoid having MI-specific logic like this in the public APIs (especially since it's quite likely that we'll need to have different types of logic for different editions in the future)